development/cpp/custom_resource_format_loaders.rst

```diff
@@ -0,0 +1,260 @@
+.. _custom_resource_format_loaders:
+
+Custom Resource Format Loaders
+==============================
+
+Introduction
+------------
+
+ResourceFormatLoader is a factory interface for loading file assets. 
+Resources are primary containers. When load is called on the same file 
+path again, the previous loaded Resource will be referenced. Naturally, 
+loaded resources must be stateless.
+
+This guide assumes the reader knows how to create C++ modules and godot 
+data types. If not, refer to this guide :ref:`doc_custom_modules_in_c++`.
+
+
+References:
+~~~~~~~~~~~
+
+- :ref:`ResourceLoader<class_resourceloader>`
+- `core/io/resource_loader.cpp <https://github.com/godotengine/godot/blob/master/core/io/resource_loader.cpp#L258>`__
+
+What for?
+---------
+
+- Adding new support for many file formats
+- Audio format 
+- Video formats
+- ML Models
+
+What not?
+---------
+
+- Raster Images
+
+Developers should use ImageFormatLoader instead
+
+References:
+~~~~~~~~~~~
+
+- `core/io/image_loader.h <https://github.com/godotengine/godot/blob/master/core/io/image_loader.h>`__
+
+
+Creating a ResourceFormatLoader
+-------------------------------
+
+Each file format consist of a data container and a ``ResourceFormatLoader``. 
+
+ResourceFormatLoaders are usually simple classes which returns all the 
+necessary metadata for godot to understanding the new extension. The 
```

Comment:
necessary metadata for **G**odot to understand the new extension (capitalize Godot, understand -> understanding)